### ISSUE TITLE
Missing documentation for seeing metrics (for debugging performance issues)

### ISSUE THREAD

**Issue by jonnor** [ipfs/go-ipfs#3177]
At certain times in our production deployment, all go-ipfs backends hits 100% CPU usage and then is unable to serve requests in a timely manner. I wish to find out where the time is spent (if possible), or in general collect metrics to figure out what could make this scenario different from when it is operating without problem.

#2881 states that there exists a (nice) metrics lib. But from searching this repository, and on Google, I cannot find a any documentation on how to access these metrics?
Any additional documentation around debugging performance issues would also be very welcomed.

--
Version: N/A
Type: question/meta
Area: documentation
Priority: P1, deployment is underperforming and slowly cooking,

**Comment by whyrusleeping** [comment]
Take a look at: https://github.com/ipfs/go-ipfs/blob/master/docs/debug-guide.md

Beyond that, there are other metrics built in, but they arent super useful for debugging performance.

**Comment by jonnor** [comment]
Thanks, that looks like a great starting point! Should this be linked in the README.md?

Some information around expected performance (some exists in various bugreports but often closed with "it is better now"), and best practices around performance.

**Comment by whyrusleeping** [comment]
@jonnor Yeah, all that would be great to have in the readme. Linking to the debug guide is a quick change, but information about expected performance and best practices is something we need to compile

**Comment by jbenet** [comment]
What metics would be useful atm? @kubuxu is working on adding some so he
could include yours

**Comment by lidel** [comment]
Closing as a lot happened since 2016, docs exist:

- https://github.com/ipfs/kubo/blob/master/docs/debug-guide.md
- https://github.com/ipfs/kubo/blob/master/docs/metrics.md
